---
title: "raptr: Representative and Adequate Prioritisation Toolkit in R"
author: |
  | Jeffrey O. Hanson$^1$, Jonathan R. Rhodes$^2$, Hugh P. Possingham$^1$, Richard A. Fuller$^1$
  | $^1$School of Biological Sciences, The University of Queensland, Brisbane, QLD, Australia
  | $^2$School of Geography, Planning and Environmental Management, The University of Queensland, Brisbane, QLD, Australia
  | Correspondance should be addressed to jeffrey.hanson@uqconnect.edu.au
date: "`r format(Sys.time(), '%d %B %Y')`"
tags: ["conservation", "biodiversity", "MILP", "reserve-selection", "optimisation", "protected area"]
abstract: "

1. An underlying aim in conservation planning is to maximise the long-term persistence of biodiversity. To fulfill this aim, the processes that sustain biodiversity must be conserved. Many biodiversity processes can be preserved by conserving a representative sample of each biodiversity feature (species or ecosystem). For example, representing ability for individuals to persist in a range of climatic conditions might be crucial for the long-term persistence of a species in a world where environmental change is accelerating. Yet current approaches overwhelmingly focus on securing a target proportion of the features' geographic range.

2. To address this issue, we developed the \\texttt{raptr: Representative and Adequate Prioritisation Toolkit in R} to guide reserve selection using explicit targets for representing biodiversity processes as well as area protection targets. Users set \"space targets\" to secure a representative sample of each feature within an attribute space that reflects a biodiversity process (eg. an attribute space expressing variation in genetic characteristics between individuals, or variation in climatic conditions between areas). Users also set \"amount targets\"--similar to conventional reserve selection methods--to secure a target proportion of the feature's geographic distribtuion. We demonstrated the functionality of this package using a simulated study and two case studies. In each study, we generated solutions that secured a proportion of the species' distributions using amount targets--representing conventional reserve selection--and compared them with solutions generated using both amount and space targets.

3. We show that explicitly considering biodiversity processes in reserve selection can substantially change the solution. The simulations suggest that including representativeness in conservation planning is important where biodiversity features have multi-model distributions in an attribute space. The case studies show that explicitly considering space-based targets for biodiversity processes can return solutions that could be far more effective in achieving long term biodiversity persistence and do not require a much greater area reserved than traditional approaches. 

4. The \\texttt{raptr R} package provides a unified framework for achieving spatial conservation prioritisations that are both adequate and representative, resulting in a greater chance of achieving long-term biodiversity persistence than area-based planning alone.

"
output:
  rmarkdown::pdf_document:
    toc: false
    keep_tex: yes
    fig_caption: yes
    includes:
      in_header: preamble.tex
fontsize: 11pt
documentclass: article
bibliography: references.bib
csl: reference-style.csl
---

```{r, include=FALSE}
# set wd
knitr::opts_knit$set(root.dir=normalizePath('../..'))
```

```{r, include=FALSE}
## initialization
# load .rda file
checkpoint::checkpoint('2016-07-25', R.version='3.3.1', scanForPackages=FALSE)
session::restore.session('data/final/results.rda')

## preliminary processing
for (i in seq_len(nlayers(cs1.bioclim.RAST)))
	cs1.bioclim.RAST@layers[[i]]@file@name <- paste0(getwd(), strsplit(cs1.bioclim.RAST@layers[[i]]@file@name, 'raptr-manuscript')[[1]][[2]])

```

<!--  title page -->
Short running title: Representative and adequate prioritisations \newline
Abstract word count: XXX / XXX  
Total word count: XXX / XXX  
Number of references: XXX  
Number of figures: XXX  
Number of tables: 0  
Number of textboxes: 0  
Number of equations: XXX  
\newpage

<!-- conservation and reserve selection -->
Perhaps the most fundamental aim of conservation is to maximise the long-term persistence of biodiversity [@r423; @r255]. To achieve this, conservation actions must preserve biodiversity patterns (eg. populations, species, ecosystems), but also crucially the processes that sustain them. One of the major tangible achievements of modern conservation has been the act of setting aside areas for preservation [@r440]. Reserve networks buffer species from gross threatening processes and set the stage for enhanced management interventions [@r465]. Since the resources available for conservation action are limited, protected area networks must be sited in places that satisfy conservation objectives for minimal cost [@r255]. However, despite the importance of biodiversity processes, they are only rarely used to guide conservation planning.

<!-- biodiversity processes are important -->
Prioritisations must capture the ecological and evolutionary processes that sustain biodiversity [@r3; @r255; @r9]. Ecological processes are important for biodiversity to persist over short time-scales. For instance, predator-prey processes can mediate ecosystem structure and function through top-down control [@r476]. Additionally, migratory processes allow many species to access suitable habitat throughout the year [@r477]. On the other hand, evolutionary processes are important for biodiversity to persist over long time-scales. Adaptive evolutionary processes foster resilience against environmental change [@r3; @r43], and neutral evolutionary processes maintain genetic diversity [@r5; @r67; @r63]. In recent decades, a wealth of data relevant to biodiversity processes has become available [eg. bioclimatic, distribution, genetic, phylogenetic, and trait data; @r406; @r33; @r460; @r453]. This is sparked significant interest in using this data to make better conservation decisions [@r68; @r9; @r473; @r247; @r448; @r468; @r407; @r61; @r2; @r470; @r475; @r469].

<!-- current methods for preserving biodiversity processes -->
Today, the most widely used conservation planning tools focus on biodiversity patterns [\texttt{Marxan} and \texttt{Zonation}; @r429; @r22]. The cornerstone of these tools is that decision makers can set targets for a proportion of the geographic range of species, or ecosystems (termed features). To incorporate some information on biodiversity processes, current methods involve partitioning the features into different groups as a pre-processing step [@r9; @r247; @r448]. These groups are partitioned based on a proxy variable that relates to a biodiversity process of interest [@r473]. For instance, species ranges' have been partitioned according to habitat discontinuities to capture adaptive processes [@r2]. However, by partitioning features into different groups, it is not clear how well a given solution samples the original proxy variable, and ultimately, how well the solution captures the biodiversity process of interest. To overcome this limitation, Faith and Walker [-@r218] developed a decision support tool (\texttt{DIVERSITY}) to explicitly sample proxy variables to secure biodiversity processes. However, this tool is not widely used today, in part, because it cannot accommodate explicit targets on biodiversity patterns for multiple features.

<!-- aims of this paper -->
One of the key issues in reserve selection is the lack of a unified decision support tool that uses explicit and separate targets for biodiversity patterns and processes. To begin to fill this gap, we united the ideas underpinning \texttt{DIVERSITY} and \texttt{MARXAN} into new formulations of the reserve selection problem, and implemented them in the \texttt{raptr R} package. This \texttt{R} package provides decision makers with the tools to generate prioritisations that preserve both biodiversity patterns and processes. Here, we aim to the provide an in-depth understanding of this \texttt{R} package and explore the functionality.

# Methods
## OPTIMISATION PROBLEM FORMULATION
Biodiversity features are defined as the entities that the prioritisation is required to preserve (eg. species, ecosystems). Spatial attributes are defined as the variation across the feature' range that the prioritisation is required to sample. They can be intrinsic (eg. genetic or trait variation) or extrinsic (eg. environmental variation) to the feature. These attributes should be related to the biodiversity processes that the decision maker aims to preserve.

A set of attributes is conceptualised as a space (termed an "attribute space"). Each planning unit may be thought to exist as a single point inside a given space. For example, a decision maker may require a prioritisation that represents populations along climatic gradients. To achieve this, the decision maker might use an "climatic" attribute space with dimensions relating to mean annual temperature ($^{\circ}$C) and precipitation (mm). Any given combination of temperature and precipitation may be conceived as a point in this environmental space. By associating planning units with climatic data, they can be mapped from geographic space to this environmental attribute space. In addition to planning units, attribute spaces also contain demand points.

Demand points are designated by the decision maker to indicate regions of the attribute space that they wish to represent in the prioritisation (see below for discussion on how demand points can be generated for real-world data sets). The amount of variation in the attribute space that a solution secures depends on the distance between each demand point and each selected planning unit in the attribute space (see below for mathematical expressions). The shorter the distances between the demand points and the planning units; the better the prioritisation is at securing the variation in the spatial attribute. To calculate the proportion of variation secured in a given solution, the distances are scaled by the distance between the demand points to their centroid (equivalent to calculating R$^2$ values for K-means clustering). In any attribute space there may exist points that are impossible (eg. mean annual rainfall -5 mm), or do not occur in the study area (eg. mean annual temperature 30$^{\circ}$C in Antarctica). Additionally, there may be some regions that are desirable for some features and undesirable for others (eg. conditions known to be outside the physiological tolerance of certain species). Thus a different set of demand points and weights are used for each attribute space and each feature. By using demand points in desirable regions of an attribute space for a given feature, the decision maker can ensure that prioritisations secure the feature in planning units with spatial attributes that are desirable for that feature.

To illustrate these concepts, consider the following example: we wish to develop a prioritisation for a single species that has four populations. However, we can only afford to preserve three of the four populations. To ensure that we preserve a representative sample, we want to want preserve populations inhabiting different environmental conditions. We obtained annual rainfall (mm) and temperature ($^{\circ}$C) data at the location of each population. We used this data to construct a two-dimensional environmental attribute space. Next, we generated demand points as equi-distant points inside this space. By comparing the distributions of demand points and populations in the attribute space, we can identify a suitable prioritisation (Fig. 1). We can see that a solution that prioritises both populations _A_ and _C_, effectively "doubles-up" on the same environmental characteristics and would contain in considerable redundancy. Instead, a more representative sample of the intra-specific variation could be captured by securing populations $A$, $B$, and $D$. This example demonstrates how attribute spaces and demand points relate to biodiversity processes, and can be used to guide reserve selection process .

The formulations used to express the reserve selection problem in the \texttt{raptr R} package are based on a combination of the \texttt{Marxan}, \texttt{DIVERSITY}, and the uncapacitated facility location problems [@r451]. Although the \texttt{raptr R} package utlises two novel formulations, for brevity, we will define the simpler formulation--referred to as the unreliable formulation--below and define the more complex version--the reliable formulation--in Appendix S1. These formulations are named after the unreliable and reliable facility location formulations from which they are based upon [@r16]. The key difference between these two formulations is that the reliable formulation explicitly considers the probability that planning units are occupied when calculating the proportion of an attribute space sampled in a solution. All mathematical terms defined hereafter are described in Table S1. For convenience, the cardinality of sets will be denoted using the same symbol used to denote the variable.

Define $F$ to be the set of features one wishes to conserve (indexed by $f$). Let $J$ be a set of planning units (indexed by $j$). Also, let $A_j$ denote the area, and $C_j$ denote the cost of preserving planning unit $j \in J$. To assess the extent to which each feature is secured in a given prioritisation, let $q_{fj}$ denote the probability of feature $f$ occupying planning unit $j$. The level of fragmentation associated with a prioritisation is parametrised as the net exposed boundary length. Let the shared edges between each planning unit $j \in J$ and $k \in J$ be $e_{jk}$.

Let $S$ denote a set of attribute spaces (indexed by $s$). Each $j \in J$ is associated with spatially explicit data that represent coordinates for each attribute space $s \in S$. Let $I_{fsi}$ denote a set of demand points (indexed by $i$) for each feature $f \in F$ and each attribute space $s \in S$. Let $\lambda_{fsi}$ denote the weighting for each demand point $i \in I$, $f \in F$ and $s \in S$. Let $d_{fsij}$ denote the distance between each demand point $i \in I$ and each planning unit $j \in J$ for each feature $f \in F$ and attribute space $s \in S$. To describe the inherent variation in the distribution of demand points for feature $f$ and space $s$, let $\delta_{fsi}$ denote the distance between each demand point $i \in I$ and the centroid of the demand points. Demand points with greater weight $\lambda_{fsi}$ are more important, and the optimal solution will be likely to select planning units close to highly weighted demand points. As a consequence, the decision maker will need to choose an appropriate weighting for each demand point.

Targets are used to ensure that prioritisations adequately preserve each feature. Amount-based targets are used to ensure that the total amount of habitat preserved is sufficient. Let $T_f$ denote the expected amount of area that needs to be preserved for each feature $f \in F$. Space-based targets ensure that a representative sample of each feature is secured. Let $\tau_{fs}$ denote the space-based targets for feature $f \in F$ and attribute space $a \in A$. For convenience, all targets are expressed as proportions in the \texttt{R} package.

The control variables for the unreliable formulation are the $B$, $T_{s}$, and $\tau_{sa}$ variables.

\begin{align*}
T_s &= \text{amount target for feature $f$} \tag*{eqn 1a}\\
%
\tau_{sa} &= \parbox{25em}{representation target for feature $f$ in attribute space $a$ according to the corresponding demand points} \tag*{eqn 1b}\\
%
B &= \text{boundary length modifier (BLM): penalise fragmented solutions} \tag*{eqn 1c}\\
\end{align*}

The decision variables are the $X_j$ and $Y_{fsij}$ variables.

\begin{align*}
X_j
	&= \begin{cases}
		1, & \parbox{25em}{if planning unit $j$ is selected for conservation action} \tag*{eqn 2a} \\
		0, & \parbox{25em}{otherwise} \\
	\end{cases} \\
%
Y_{fsij} &= \begin{cases}
		1, & \parbox{25em}{if demand point $i$ is used to represent planning unit $j$ for feature $f$ in space $s$. } \tag*{eqn 2b} \\
		0, & \parbox{25em}{otherwise} \\
	\end{cases} \\
\end{align*}

Each demand point $i \in I$ for feature $f \in F$ and space $s \in S$ is assigned to a selected planning unit $J$ where $X_j = 1$. The weighted distance between the demand point and its assigned planning unit $\lambda_{fsi} d_{fsij}$ is used to assess how well the demand point is represented in a given solution. Generally, demand points are assigned to the closest selected planning units (unless particularly low space-based targets are used).

The unreliable formulation (URAP) is a defined as a multi-objective optimisation problem.

\begin{align*}
& \text{(URAP)} & \text{Min } & \sum_{j=0}^{J-1} \left( X_j C_j \right) + \sum_{j=0}^{J-1} \sum_{k=j}^{J-1} X_j \left( 1-X_k \right) \left( B e_{jk} \right) + & & \tag*{eqn 3a} \\
%
& & \text{s.t. } & \sum_{j=0}^{J-1} A_j q_{fj} \geq T_{f} & \forall & 0 \leq f \leq F-1 \tag*{eqn 3b}\\
%
& & & 1 - \frac{\sum_{i=0}^{I-1} \sum_{j=0}^{J-1} \lambda_{fsi} {d_{fsij}}^{2} Y_{fsij}}{\sum_{i=0}^{I-1} \lambda_{fsi} {\delta_{fsi}}^{2}} \geq \tau_{fs} & \forall & 0 \leq f \leq F-1, \tag*{eqn 3c}\\
& & & & & 0 \leq s \leq S-1\\
%
& & & \sum_{j=0}^{J-1} Y_{fsij} = 1 & \forall & 0 \leq f \leq F-1, \tag*{eqn 3d}\\
& & & & & 0 \leq s \leq S-1, \\
& & & & & 0 \leq i \leq I-1\\
%
& & & Y_{fsij} \leq X_j & \forall & 0 \leq f \leq F-1, \tag*{eqn 3e}\\
& & & & & 0 \leq s \leq S-1, \\
& & & & & 0 \leq i \leq I-1,\\
& & & & & 0 \leq j \leq J-1\\
%
& & & X_j, Y_{fsij} \in {0,1} & \forall & 0 \leq f \leq F-1, \tag*{eqn 3f}\\
& & & & & 0 \leq s \leq S-1,\\
& & & & & 0 \leq i \leq I-1\\
%
\end{align*}

The objective function (eqn 3a) determines the utility of a given prioritisation: a combination of the total cost of a prioritisation and how fragmented it is. Constraints (eqn 3b) ensure that all the amount-based targets are met. Constraints (eqn 3c) ensure that all the space-based targets are met for each feature and each attribute space. For each feature and attribute space, the total weighted distance between the demand points and their closest selected planning units is calculated ($\lambda_{fsi} d_{fsij} Y_{fsij}$). This total weighted distance is then scaled by the inherent variation in the demand points ($\lambda_{fsi} \delta_{fsi}$). The resulting fraction yields a proportion conceptually similar to the $R^2$ statistic used in $k$-means clustering analysis. The constraints ensure that this proportion must be equal to or greater than the space-based target. Constraints (eqn 3d) ensure that only one planning unit is assigned to each demand point. Constraints (eqn 3e) ensure that demand points are only assigned to selected planning units. Constraints (eqn 3f) ensure that the $X$ and $Y$ variables are binary.

## OPTIMISATION
The unreliable formulation is non-linear. However, the non-linear components can be linearised using existing techniques. The expression $X_j X_k$ in (eqn 3a) can be linearised using methods described by Beyer et al. [-@r426]. Linearised versions of the problems can be solved using commercial exact algorithm solvers. The \texttt{raptr} R package provides functions to express conservation planning data as optimisation problems using linearised versions of the unreliable and reliable formulations (see Appendix S1 for formulation). These optimisation problems can then be solved to generate prioritisations using the commercial \texttt{Gurobi} software suite (\url{http://www.gurobi.com}). Presently, academics can obtain a license at no cost from the Gurobi website.

# Examples

To understand the behaviour of the unreliable problem and showcase its value, we conducted a simulation study and two case studies. These studies involved generating solutions using only amount targets to represent prioritisations generated using conventional methods (eg. \texttt{Marxan}), and solutions using amount and space targets using the unreliable formulation. By comparing these solutions, we can guage the benefits of explicitly including space targets in reserve selection. All analyses were performed in \texttt{R} [version `r getRversion()`; @r192].

## SIMULATION STUDY
### Methods
We simulated a hypothetical study area with planning units arranged in a $`r sqrt(sim.params.LST[[MODE]]$number.planning.units)` \times `r sqrt(sim.params.LST[[MODE]]$number.planning.units)`$ square grid (Fig. 2). We then simulated three species across this study area. The first species was simulated to represent a hyper-generalist--occurring in all planning units with equal probability (Fig. 2a). This species' distribution was based on a uniform distribution (eqn 8a). The second species was simulated to represent a species with an idealised distribution (Fig. 2b). It has a core range area, and is less likely to be found in areas that are distant from the core area. This species' distribution was simulated using the probability density function of a single multivariate normal distribution (eqn 8c). The third species represents a species with two distinct populations (Fig. 2c). This species' distribution was simulated to depict a bimodal distribution, based on the combination of the probability density functions of two multivariate normal distributions (eqn 8d). The probability that each species inhabited a given planning unit was calculated using the $X, Y$ coordinates of the planning unit and eqns 8a--8d. We used a geographic attribute space to showcase the behavior of the problem. For each species, demand points were generated by calculating the centroids of each planning units and their weights were set as the probability that the planning units were occupied. 

\begin{align*}
\\
P \left( \text{uniformly distributed species} | \left( x, y \right) \right) &= 0.1 \tag*{eqn 8a} \\
%
f \left(z, \mu, \Sigma \right) &= \left(2 \pi \right) ^{-1} | \Sigma | e ^{- \frac{1}{2} \left(z - \mu \right)' \Sigma ^{-1} \left(z - \mu \right)} \tag*{eqn 8b} \\
%
P \left( \text{normally distributed species} | \left( x , y \right) \right) &= \frac{f \left( \left[ \begin{smallmatrix} x \\ y \end{smallmatrix} \right] \, \left[ \begin{smallmatrix} 0.0 \\ 0.0 \end{smallmatrix} \right] \, \left[ \begin{smallmatrix} 12.58&0 \\ 0&12.5 \end{smallmatrix} \right] \right)}{2} \tag*{eqn 8c} \\
%
P \left( \text{bimodally distributed species} | \left( x,y \right) \right) &= \text{Max} \begin{cases}
f \left(\left[ \begin{smallmatrix} x \\ y \end{smallmatrix} \right] \, \left[ \begin{smallmatrix} -3.75 \\ -3.75 \end{smallmatrix} \right] \, \left[ \begin{smallmatrix} 10&0 \\ 0&10 \end{smallmatrix} \right] \right), \\ \frac{f \left( \left[ \begin{smallmatrix} x \\ y \end{smallmatrix} \right] \, \left[ \begin{smallmatrix} 3.75 \\ 3.75 \end{smallmatrix} \right] \, \left[ \begin{smallmatrix} 8&0 \\ 0&8 \end{smallmatrix} \right] \right)} {2} \\
\end{cases} \tag*{eqn 8d} \\
%
\end{align*}

We generated four solutions for each species. First to represent solutions generated using conventional conservation planning methods, we generated solutions using only `r sim.params.LST[[MODE]]$amount.target*100` % amount targets. Second to show how the addition of geographic targets can affect a prioritisation, we generated solutions using `r (sim.params.LST[[MODE]]$amount.target*100)` % amount targets and `r (sim.params.LST[[MODE]]$space.target*100)` % geographic targets. Third to represent solutions using conventional planning methods that penalise for fragmentation, we generated solutions using `r (sim.params.LST[[MODE]]$amount.target*100)` % amount targets and a boundary length modifier of `r sim.params.LST[[MODE]]$blm`. Fourth to illustrate the combined affect using geographic targets and penalising fragmentation, we generated solutions using using `r (sim.params.LST[[MODE]]$amount.target*100)` % amount targets and `r (sim.params.LST[[MODE]]$space.target*100)` % geographic targets and a boundary length modifier of `r sim.params.LST[[MODE]]$blm`.

### Uniformly distributed species

The uniform species was simulated to occur with a constant probability of occupancy across the study area (Fig. 2a). The solution generated using `r sim.params.LST[[MODE]]$amount.target*100` % amount-based targets (Fig. 3a) selected `r filter(sim.spp.DF, species=='Uniform\nspecies', prioritisation=='Amount\ntargets')$n` planning units near the southern end of study area. This specific configuration is an artifact of the \texttt{Gurobi} solver used to solve the problem. In fact, for this species, all prioritisations containing `r filter(sim.spp.DF, species=='Uniform\nspecies', prioritisation=='Amount\ntargets')$n` planning units are optimal when considering only `r sim.params.LST[[MODE]]$amount.target*100` % amount targets. In the absence of criteria to guarantee representativeness, reserve selection methods may or may not return solutions that secure a representative sample of the features. In this particular case, the solution does not secure a representative sample of the species' range (`r round(filter(sim.spp.DF, species=='Uniform\nspecies', prioritisation=='Amount\ntargets')$space.held,2)` % sampled). Note that the proportion here is negative because it performs worse than simply siting a planning unit in the centroid of the species' demand points.

The addition of a `r sim.params.LST[[MODE]]$space.target*100` % geographic target resulted in a solution that secured a representative sample of the uniform species' range (`r round(filter(sim.spp.DF, species=='Uniform\nspecies', prioritisation=='Amount &\nspace targets')$space.held,2)` % sampled; Fig. 3b). Since all planning units have equal cost and an equal chance of being occupied, this solution has the same number of planning units as the solution generated using only amount targets (cf. Fig 3a). Although the use of amount and space based targets has addressed adequacy and representativeness objectives (respectively), they have resulted in a highly fragmented solution.

The addition of a positive boundary length modifier (BLM) parameter resulted in a well-connected solution that secured an adequate proportion (`r round(filter(sim.spp.DF, species=='Uniform\nspecies', prioritisation=='Amount &\nspace targets')$amount.held,2)` % ) and representative sample (`r round(filter(sim.spp.DF, species=='Uniform\nspecies', prioritisation=='Amount & space\ntargets and BLM')$space.held,2)` % sampled) of the species geographic distribution (Fig. 3d). This solution contains `r filter(sim.spp.DF, species=='Uniform\nspecies', prioritisation=='Amount & space\ntargets and BLM')$n` planning units--a few more than the previously discussed solutions--to ensure that the solution secures a representative proportion of the species geographic distribution in a configuration that is not highly fragmented. This result suggests that the combination of space targets and boundary length modifiers may yield solutions that contain more planning units, since the solution generated using amount targets and a boundary length modifier (Fig. 3c) contained the sample number of planning units as the solution with just the amount targets (Fig. 3a). Overall, these results show that under the simplest of conditions, the reliance on just amount targets can cause an under-specified reserve selection problem that is unlikely to return a suitable solution for implementation.

### Normally distributed species

The normally distributed species was simulated to contain a single core area where individuals are most prevalent and marginal areas where individuals are less likely to occur (Fig. 2b). The solution generated using `r sim.params.LST[[MODE]]$amount.target*100` % amount targets contained `r filter(sim.spp.DF, species=='Normal\nspecies', prioritisation=='Amount\ntargets')$n` planning units and concentrated conservation efforts in the core area (Fig. 3e). Since all planning units have equal costs and areas, this solution contains the planning units with the highest probabilities of occupancy. Whilst this solution satisfies the adequacy objective for a prioritisation in a cost-effective manner--it fails to fulfill the representative objective for a prioritisation (`r round(filter(sim.spp.DF, species=='Normal\nspecies', prioritisation=='Amount\ntargets')$space.held,2)` % distribution sampled).

The solution generated using `r sim.params.LST[[MODE]]$amount.target*100` % amount targets and `r sim.params.LST[[MODE]]$space.target*100` % geographic targets resulted in a solution that is both adequate and representative in terms of the uniform species distribution (Fig. 3f). This solution used `r filter(sim.spp.DF, species=='Normal\nspecies', prioritisation=='Amount &\nspace targets')$n` planning units to secure `r round(filter(sim.spp.DF, species=='Normal\nspecies', prioritisation=='Amount &\nspace targets')$amount.held,2)` % and sample `r round(filter(sim.spp.DF, species=='Normal\nspecies', prioritisation=='Amount &\nspace targets')$space.held,2)` % of the normal species' distribution. By using amount and geographic targets in the reserve selection problem, we have obtained a solution that concentrates conservation effort in the range core and also the range margin. However, similar to the corresponding solution for the uniform species (Fig. 3b), this solution is highly fragmented.

By using `r sim.params.LST[[MODE]]$amount.target*100` % amount targets and `r sim.params.LST[[MODE]]$space.target*100` % geographic targets and a boundary length modifier parameter ($BLM = 1$), we obtained a solution that fulfills adequacy, representativeness and connectivity objectives (Fig. 3h). This solution contains `r filter(sim.spp.DF, species=='Normal\nspecies', prioritisation=='Amount & space\ntargets and BLM')$n` planning units. Similar to the corresponding solution for the uniform species (Fig. 3d), this solution contains more planning units than any other solutions for this species (cf. Figs. 3e--3g). The results for the normally distributed species suggest that the space targets can result in solutions that secure a more representative sample of the species'--even for species without significant spatial structure.

### Bimodally distributed species

The bimodally distributed species was simulated to represent a species with a highly structured distribution, and it contains two distinct populations (Fig. 2c). The solution generated using just `r sim.params.LST[[MODE]]$amount.target*100` % amount targets assigned conservation effort to just one of the two populations (Fig. 3i). While this solution secured an adequate proportion of the species' distribution (`r round(filter(sim.spp.DF, species=='Bimodal\nspecies', prioritisation=='Amount\ntargets')$amount.held,2)` % secured), it did not sample a representative proportion of the range (`r round(filter(sim.spp.DF, species=='Bimodal\nspecies', prioritisation=='Amount\ntargets')$space.held,2)` % sampled). The addition of boundary length modifiers to just amount targets resulted in a solution which sampled even less of the species distribution (`r round(filter(sim.spp.DF, species=='Bimodal\nspecies', prioritisation=='Amount target\nand BLM')$space.held,2)` % sampled; Fig. 3k). However, the addition of geographic space targets resolved these issues.

The solution generated using `r sim.params.LST[[MODE]]$amount.target*100` % amount and `r sim.params.LST[[MODE]]$space.target*100` % geographic targets included planning units from both populations (Fig. 3j). Although this solution required more planning units to fulfill both targets ($n=`r filter(sim.spp.DF, species=='Bimodal\nspecies', prioritisation=='Amount &\nspace targets')$n`$), this solution secured a representative sample of the bimodally distributed species (`r round(filter(sim.spp.DF, species=='Bimodal\nspecies', prioritisation=='Amount &\nspace targets')$space.held,2)` % range sampled). Similar to corresponding solutions generated for both the uniformly (Fig. 3b) and normally distributed species (Fig. 3f), this solution was also highly fragmented and we could obtain a more well-connected solution at the expense of selecting additional planning units ($n=`r filter(sim.spp.DF, species=='Bimodal\nspecies', prioritisation=='Amount & space\ntargets and BLM')$n`$; Fig. 3l). The results for the bimodally distributed species suggest that species with highly structured populations or strong variation between individuals could benefit the most from prioritisations that are generated using space-based targets.

## CASE STUDY 1
### Methods
We investigated how space-based targets can be used in a multi-species planning context to generate a prioritisation that sufficiently preserves the realised niche for several species. By preserving the populations in suitable habitats with different environmental conditions, conservation planners can preserve the species' adaptive landscape and foster resilience against environmental change [@r63]. We selected Queensland, Australia as the study area. This region is ideal for exploring the potential of niche-based targets because it contains a broad range of different habitats. We obtained data for 19 bioclimatic variables across the region [at $30^{\prime \prime}$ resolution from \url{www.worldclim.org}; @r33] and subjected them to a principal components analysis (using ArcMap 10.3.1). We used the first `r as.character(as.english(cs1.params.LST[[MODE]]$n.pc.layers))` principal components (cumulatively explained `r round(cs1.pca.DF[[4]][cs1.params.LST[[MODE]]$n.pc.layers],1)` % of the total variation) to characterise the environmental variation across the study area (Fig. 4).

We selected four bird species that span a range of different evolutionary histories, distributions and ecologies: blue-winged kookaburra (_Dacelo leachii_), brown-backed honeyeater (_Ramsayornis modestus_), brown falcon (_Falco berigora_), and pale headed rosella (_Platycercus adscitus_). We then mapped the extent of occurrence for each species (Fig. 5). To do this, we obtained observation data from the Atlas of Living Australia across the whole of Australia [using the \texttt{ALA4R R} package; @r453], spatially thinned the data to omit points within `r cs1.params.LST[[MODE]]$thin.distance/1000` km$^2$ of each other to ameliorate the effects of sampling bias [using the developmental version of the \texttt{spThin R} package; \url{www.github.com/mlammens/spThin}; @r454], and fit `r cs1.params.LST[[MODE]]$mcp.percent` % minimum convex polygons [using the \texttt{adehabitatHR R} package; @r455]. We used this method because it is entirely reproducible using freely available data. 

We generated `r cs1.params.LST[[MODE]]$dp.number` demand points for each species (Fig. S2). We used the following method to ensure that the demand points reflected the core parts of the species' realised niches. For each species, we generated random points inside the species range and at each point extracted the principal component values at that location. We then fit hyperbox kernels to the distribution of principal component values to characterise the realised niche of each species [using a manually chosen bandwidth of `r cs1.params.LST[[MODE]]$dp.bandwidth[1]` and a `r cs1.params.LST[[MODE]]$dp.quantile` quantile to map the core parts of the species' niches; implemented in the \texttt{hypervolume R} package; @r231]. We then generated uniformly distributed points inside the species' kernels and estimated the density of the training points at the uniformly generated points. These uniformly distributed points and associated density estimated were used as demand point coordinates and weights (respectively).

We generated two solutions. The first solution was generated using `r cs1.params.LST[[MODE]]$amount.target*100` % amount-based targets for all species. The second solution was generated the same amount-based target with an additional `r cs1.params.LST[[MODE]]$space.target*100` % niche-based target for each species. 

### Results
Generally, the solution generated using amount targets preserved a representative sample of each the `r as.character(as.english(length(cs1.params.LST[[MODE]]$species.names)))` bird species' niches--with `r ifelse(length(amount.not.represented.species.names)>1, as.character(as.english(length(amount.represented.species.names))), 'one')` excpetion. Specifically, this solution sampled over `r cs1.params.LST[[MODE]]$space.target*100` % of the realised niche of `r parsed.representative.space.held.names`. Yet it failed to achieve this for `r parsed.not.representative.space.held.names`. This result demonstrates that though solution generated using conventional methods may preserve a representative sample of some features (Figure S1), only through the use of explicit targets can decision makers be sure that solutions secure a representative sample for all features.

## CASE STUDY 2
### Methods
Here we used space-based targets to generate a prioritisation securing a representative sample of a species' intra-specific genetic variation. We used species occurrence and multilocus AFLP data collected by the international IntraBioDiv project in the European Alps [@r463; @r461; @r462]. Although this dataset contains multiple plant species, we used data for the `r tolower(cs2.params.LST[[MODE]]$common.name)` (\textit{`r gsub('_', ' ',cs2.params.LST[[MODE]]$species.name,fixed=TRUE)`}), a wide-spread species with significant genetic structure [@r462] as a simple case study. Members of the IntraBioDiv project collected data using a $20^{\prime}$ longitude by $21^{\prime}$ latitude grid (approx. 22.3 km $\times$ 25 km; Fig. 7a). They visited every second grid cell, and if the `r tolower(cs2.params.LST[[MODE]]$common.name)` was detected in a cell, samples were collected from three individuals. Samples were genotyped using amplified fragment length polymorphisms [AFLP; @r472], and used to construct matrices denoting the presence/absence of polymorphisms at loci. In total, `r structurer::n.samples(cs2.spp.StructureData.LST[[1]])` individuals were genotyped at `r structurer::n.loci(cs2.spp.StructureData.LST[[1]])` markers.

We used non-metric multi-dimensional scaling [NMDS; using Gower distances to accommodate sparsity; @r459; implemented the \texttt{cluster R} package; @r457] to ordinate the presence (or absence) of locus-specific alleles wthin individuals into two continuous variables [implemented in the \texttt{vegan R} package; @r458]. These continuous variables described the main axes of genetic variation within the species. We calculated the average of the values associated with individuals in each grid cell. These values were used to create a genetic attribute space (Figs. 7c and 7d). To assess spatial auto-correlation, we calculated Moran's I auto-correlation index for each NMDS axis using inverse great circle distances based on the grid cells' centroids [using the \texttt{ape R} package; @r464].

The grid cells were used as planning units. The grid-cell averaged ordinations were used to describe the typical genetic characteristics of individuals in the planning unit. Since the number of planning units was relatively small, we used the same grid-cell averaged ordinations as demand points. To ensure that the solutions did not prioritise particularly costly areas, we included opportunity cost data (Fig. 7b). We obtained population density data [1 km resolution from the Global Rural-Urban Mapping Project; \texttt{GRUMP V1}; @r456] and estimated the total population density inside each grid cell.

We generated two solutions. The first solution was generated using an `r cs2.params.LST[[MODE]]$amount.target*100` % amount-based target. The second solution was generated using the same amount-based target and an additional `r cs2.params.LST[[MODE]]$genetic.target*100` % genetic-based target.

### Results

A two-dimensional attribute genetic space was able to capture most of the variation between individuals (stress = `r round(cs2.spp.nmds.LST[[1]]$stress,2)`; Figs. 7c--7d). Planning units located near each were contained individuals with similar genetic characteristics (Moran's I: NMDS axis 1, $I = `r round(cs1.nmds.MoransI[[1]][[1]]$observed,2)`$, $P `r pvalString(cs1.nmds.MoransI[[1]][[1]]$p.value)`$; Moran's I: NMDS axis 2, $I = `r round(cs1.nmds.MoransI[[1]][[2]]$observed,2)`$, $P `r pvalString(cs1.nmds.MoransI[[1]][[2]]$p.value)`$). In terms of the average genetic characteristics of individuals in the planning units, they tended to cluster into two main groups, with evidence of within-group structure inside the larger group (Fig 8c). This analysis supports previous work by Alvarez _et al._ [-@r461] who also found evidence of genetic structure within this species.

The solution generated using just the amount target failed to preserve a representative portion of the species' genetic variation (`r round(cs2.spp.DF$genetic[1],2)` % sampled; Figs. 8a and 8c). This solution only sampled planning units that contained individuals belonging to one of the main two groups. On the other hand, the solution generated using additional genetic targets selected planning units belonging to both of the main groups. Note that since these solutions were generated using opportunity cost data, the solution generated using only the amount target is essentially the minimum number of least costly areas needed to fulfill the amount target. The only difference between the two solutions is a single planning unit. By swapping this single planning unit, the solution generated using amount and genetic targets was able to preserve a representative portion of the species' genetic variation (`r round(cs2.spp.DF$genetic[2],2)` % sampled; Figs. 8b and 8d) for only a minor increase in cost (`r round(cs2.spp.DF$Score[1],2)` total cost compared to `r round(cs2.spp.DF$Score[2],2)` total cost).

# Implications and future directions
The \texttt{raptr} R package provides a unified approach to reserve selection. One of the key advantages of this package is that it is general enough to incorporate any spatial variable as an attribute space. This package can accommodate intrinsic or extrinsic variation to the feature(s). For example, adaptation processes could be secured using environmental variation or genetic variation among loci under selection. Ecological processes, such as predator-prey interactions, could be secured by capturing intra-specific trophic variation. Additionally, different attribute spaces with different targets can be specified for different species. As long as the variation can be described using Euclidean distances, the \texttt{R} package can be used to obtain a representative sample (note that data may require transformation or dimension reduction to meet this assumption). This \texttt{R} package provides decision makers with the tools to generate prioritisations that secure both biodiversity patterns and the processes that maintain biodiversity. Additionally, the package contains functionality to accommodate uncertainty in the distribution of features, and also identify suitably connected reserves. Both the simulated and case study species suggest that explicitly considering proxies for biodiversity processes can significantly alter prioritisations.

The degree to which a prioritisation truly secures a representative sample of a feature depends on the attribute spaces and distribution of demand points chosen by the decision maker. Ultimately, the space-based targets are set as a proportion based on the distribution of the demand-points. As a consequence, if the decision maker uses an inappropriate set of spatial variables to construct an attribute space, or an inappropriate set of demand points , then the optimal solution will not be an effective prioritisation. We therefore stress that decision makers must carefully consider which biodiversity processes need to be reflected in the prioritisation, and which spatial data can be used to represent these processes. Note that maximising one process can be detrimental to another. For instance, to maximise the geographic spread of a prioritisation, reserves need to be further away from each other, yet to maximise the connectivity of a prioritisation, reserves need to be closer to each other. To assist in the selection of appropriate demand points, the \texttt{R} package provides several routines for generating demand points (see the `make.DemandPoints` function). These routines essentially use the distribution of a feature in the attribute space to define a polygon. Demand points are then generated as random points within the polygon. A kernel is then fit to the distribution of the feature in the space [using @r231; @r450], and the demand points are weighted based on the estimated density of the feature at the demand points.

The formulation requires spatially comprehensive data to map attribute spaces to planning units. For instance, in both case-studies, all planning units occupied by the species were associated with coordinates in the attribute spaces. However, most real-world data sets are patchy--some planning units will be occupied by species for which attribute space data is not available. To use such patchy data with the \texttt{raptr R} package, the gaps in the data must first be filled in. Spatially explicit models [eg. krigging or generalised dissimilarity models; @r467; @r466] could be used to estimate data for locations that are missing from the data set as a pre-processing step. This approach has been successfully applied to a range of biological data sets [eg. @r468].

To maximise the long-term persistence of biodiversity, decision makers need to identify prioritisations that preserve existing patterns of biodiversity and the processes that support them. To achieve this, conservation planners need a decision support tool that can explicitly accommodate biodiversity patterns and processes. Here, we developed the \texttt{raptr R} package to fill this void. By exploring the functionality of this package using several simulated and case study species, we found that including space-based targets can result in substantially better solutions.

# Acknowledgements
JOH is funded by an Australian Postgraduate Award (APA) scholarship. RAF has an Australian Research Council Future Fellowship. This work was supported by the Centre of Biodiversity and Conservation Science (CBCS).

# Data accessibility
The \texttt{raptr R} package can be downloaded from The Comphrehensive R Archive Network (\url{https://cran.r-project.org/}). All data, code, and results are stored in an online repository (\url{https://www.github.com/jeffreyhanson/raptr-manuscript}) to permit replication and validation of this study.

# References

\bibliography{references}
